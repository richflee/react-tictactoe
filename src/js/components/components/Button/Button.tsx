// Button.tsx
import * as React from 'react';


const STATUS = {
    HOVERED: 'hovered',
    NORMAL: 'normal',
};

interface ButtonComponentState {
    class: string
}

interface ButtonComponentProps {
}

export default class Button extends React.Component<ButtonComponentProps, ButtonComponentState> {
    constructor(props: ButtonComponentProps) {
        super(props);

        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {
            class: STATUS.NORMAL,
        } as ButtonComponentState;
    }

    _onMouseEnter() {
        this.setState({ class: STATUS.HOVERED });
    }

    _onMouseLeave() {
        this.setState({ class: STATUS.NORMAL });
    }

    render() {
        return (
            <button type="button"
                className={this.state.class}
                onMouseEnter={this._onMouseEnter}
                onMouseLeave={this._onMouseLeave}>
                {this.props.children}
            </button>
        );
    }
}