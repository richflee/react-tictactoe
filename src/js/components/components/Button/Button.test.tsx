// Link.react.test.js
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Button from './Button';

test('Button changes the class when hovered', () => {
    const component = renderer.create(
        <Button>Foo</Button>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});