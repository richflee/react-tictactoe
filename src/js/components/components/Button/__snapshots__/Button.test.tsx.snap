// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes the class when hovered 1`] = `
<button
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Foo
</button>
`;

exports[`Button changes the class when hovered 2`] = `
<button
  className="hovered"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Foo
</button>
`;

exports[`Button changes the class when hovered 3`] = `
<button
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Foo
</button>
`;
